test/input_gen/genInput.py

```diff
@@ -93,22 +102,73 @@ def conv2d_tf(x, kernel, batch, width, height, channel, k_width, k_height, k_num
     bias_w = tf.constant_initializer(bias)
     conv2d_layer = tf.keras.layers.Conv2D(k_num, k_width, strides = stride, padding=pad, kernel_initializer=kernel_w, bias_initializer=bias_w)(tf_input)
 
-    conv2d_variables = tf.compat.v1.get_collection(tf.GraphKeys.GLOBAL_VARIABLES)
+    optimizer = tf.keras.optimizers.SGD(learning_rate = 1)
 
-    input_variables = [tf_input] + conv2d_variables
+    trainable_variables = tf.compat.v1.trainable_variables()
+    all_variables = [tf_input] + trainable_variables
 
-    grad = tf.gradients(conv2d_layer, input_variables)
+    grad = tf.gradients(conv2d_layer, all_variables)
+    train_op = optimizer.apply_gradients(list(zip(grad[1:], trainable_variables)))
 
     with tf.compat.v1.Session() as sess:
         sess.run(tf.global_variables_initializer())
-        conv2d_result = sess.run(conv2d_layer, feed_dict={tf_input: x})
-        grad_result = sess.run(grad, feed_dict={tf_input:x})
+
+        conv2d_result, grad_result, _ = sess.run([conv2d_layer, grad, train_op], feed_dict={tf_input: x})
```

Comment:
take an input for number of iterations. Then run this in a loop.
This allows the earlier unittest - `conv2d_1` and `conv2d_2` which also use this function to be reproducible

Reply:
fixed. thanks.